cpp/src/plasma/common/io.cc

```diff
@@ -0,0 +1,411 @@
+#include "io.h"
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <string.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include <stdarg.h>
+#include <sys/ioctl.h>
+#include <netinet/in.h>
+#include <utstring.h>
+#include <netdb.h>
+
+#include "common.h"
+#include "event_loop.h"
+#include "arrow/util/logging.h"
+
+#ifndef _WIN32
+/* This function is actually not declared in standard POSIX, so declare it. */
+extern int usleep(useconds_t usec);
+#endif
+
```

Comment:
This file is definitely needed in Ray as well, so what's the plan for that? Should we just duplicate it?